Sources/MapboxMaps/Foundation/Camera/BasicCameraAnimator.swift

```diff
@@ -0,0 +1,203 @@
+    public var isReversed: Bool { propertyAnimator.isReversed }
+
+    /// A Boolean value that indicates whether a completed animation remains in the active state.
+    public var pausesOnCompletion: Bool {
+        get { propertyAnimator.pausesOnCompletion}
+        set { propertyAnimator.pausesOnCompletion = newValue }
+    }
+
+    /// Value that represents what percentage of the animation has been completed.
+    public var fractionComplete: Double {
+        get { Double(propertyAnimator.fractionComplete) }
+        set { propertyAnimator.fractionComplete = CGFloat(newValue) }
+    }
+
+    // MARK: Initializer
+    internal init(delegate: CameraAnimatorDelegate,
+                  propertyAnimator: UIViewPropertyAnimator,
+                  owner: AnimationOwner,
+                  cameraView: CameraView = CameraView()) {
+        self.delegate = delegate
+        self.propertyAnimator = propertyAnimator
+        self.owner = owner
+        self.cameraView = cameraView
+    }
+
+    deinit {
+        propertyAnimator.stopAnimation(false)
+        propertyAnimator.finishAnimation(at: .current)
+        cameraView.removeFromSuperview()
+    }
+
+    /// Starts the animation if this animator is in `inactive` state. Also used to resume a "paused" animation.
+    public func startAnimation() {
+
+        if self.state != .active {
+
+            guard let delegate = delegate else {
+                fatalError("CameraAnimator delegate cannot be nil when starting an animation")
+            }
+
+            guard let animation = animation else {
+                fatalError("Animation cannot be nil when starting an animation")
+            }
+
+            // Set up the short lived camera view
+            delegate.addViewToViewHeirarchy(cameraView)
+
+            var cameraTransition = CameraTransition(with: delegate.camera, initialAnchor: delegate.anchorAfterPadding())
+            animation(&cameraTransition)
+
+            propertyAnimator.addAnimations { [weak cameraView] in
+                guard let cameraView = cameraView else { return }
+                cameraView.syncLayer(to: cameraTransition.toCameraOptions) // Set up the "to" values for the interpolation
+            }
+
+            cameraView.syncLayer(to: cameraTransition.fromCameraOptions) // Set up the "from" values for the interpoloation
+            transition = cameraTransition // Store the mutated camera transition
+        }
+
+        propertyAnimator.startAnimation()
+    }
+
+    /// Starts the animation after a delay
+    /// - Parameter delay: Delay (in seconds) after which the animation should start
+    public func startAnimation(afterDelay delay: TimeInterval) {
+        delayedAnimationTimer = Timer.scheduledTimer(withTimeInterval: delay, repeats: false, block: { [weak self] (timer) in
+            guard let self = self else { return }
+            self.startAnimation()
+            timer.invalidate()
```

Comment:
The timer has already fired, so no need to invalidate it

Reply:
This is done.